task_three_and_four/src/test/java/com/epam/webdriver/aleksandr_gostev/page/BasePage.java

```diff
@@ -0,0 +1,43 @@
+package com.epam.webdriver.aleksandr_gostev.page;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+import java.util.List;
+
+public class BasePage {
+
+    private final WebDriver driver;
+
+    public BasePage(WebDriver driver) {
+        this.driver = driver;
+        PageFactory.initElements(driver, this);
+    }
+
+    public WebDriver getDriver() {
+        return driver;
+    }
+
+    public WebElement waitForVisibility(WebElement webElement) {
+        return new WebDriverWait(driver, Duration.ofSeconds(10), Duration.ofSeconds(5))
+                .until(ExpectedConditions
+                        .visibilityOf(webElement));
+    }
+
+    public List<WebElement> waitForPresence(String xpath) {
```

Comment:
It would be better to pass By instead of String, it would allow us to wait for other locator strategies